control_panel_api/services.py

```diff
@@ -1,10 +1,33 @@
+import logging
+
+from botocore.exceptions import ClientError
 from django.conf import settings
 
 from . import aws
 
 READWRITE = 'readwrite'
 READONLY = 'readonly'
 
+logger = logging.getLogger(__name__)
+
+
+def ignore_existing(func):
+    """Decorates a function to catch and allow exceptions that are thrown for
+    existing entities or already created buckets etc, and reraise all others
+    """
+    e_names = 'BucketAlreadyOwnedByYou', 'EntityAlreadyExistsException'
```

Comment:
This should be `exception_names` or maybe just `exceptions`. Also, if it's a tuple, please include the parentheses.

Reply:
I'm assuming this isn't a list of the actual exception classes because boto dynamically creates them?

Reply:
👍

Reply:
Yep sorry should have commented - because they're dynamic I have to catch the parent then check their names.